packages/paypal-commerce-integration/src/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -91,16 +105,45 @@ export default class PayPalCommercePaymentStrategy implements PaymentStrategy {
             throw new PaymentArgumentInvalidError(['payment']);
         }
 
+        const { paymentData } = payment;
+        const isVaultedFlow =
+            paymentData && isVaultedInstrument(paymentData) && isHostedInstrumentLike(paymentData);
+
         if (!this.orderId) {
-            throw new PaymentMethodInvalidError();
+            if (isVaultedFlow) {
+                this.orderId = await this.createOrder();
+            } else {
+                throw new PaymentMethodInvalidError();
+            }
+        }
```

Comment:
It might be better to rewright this part with 2 different conditions:
```suggestion
            if (!this.orderId && isVaultedFlow) {
                this.orderId = await this.createOrder();
            }
            
            if (!this.orderId) {             
                throw new PaymentMethodInvalidError();
            }
```